### PR TITLE
[15.0][IMP] fleet_vehicle_inspection: Add cost vehicle inspection and create vehicle service

### PR DESCRIPTION
**Pull request by carolinafernandez-tecnativa**
- Add Cost vehicle inspections. Once is confirmed, service vehicle is created. If inspection is cancelled, service vehicle is deleted.

TT46534

@Tecnativa

@pedrobaeza @pilarvargas-tecnativa

### CODE REVIEW

**Review by marcelsavegnago — APPROVED** [review]

**Review by pilarvargas-tecnativa — APPROVED** [review]

**Comment by marcelsavegnago** [comment]
ping @ivantodorovich

File: fleet_vehicle_inspection/models/fleet_vehicle_inspection.py [hunk 1]
```diff
@@ -132,6 +148,9 @@ def create(self, vals):
 
     def button_cancel(self):
         records = self.filtered(lambda rec: rec.state in ["draft", "confirmed"])
+        services = records.filtered(lambda rec: rec.service_id).mapped("service_id")
```
Comment (ivantodorovich) [inline fleet_vehicle_inspection/models/fleet_vehicle_inspection.py]:
```suggestion
        services = records.service_id
```

💡 tip: This is exactly the same thing, in only 1 step

File: fleet_vehicle_inspection/models/fleet_vehicle_inspection.py [hunk 2]
```diff
@@ -145,6 +164,25 @@ def button_confirm(self):
...
+                        "service_type_id": self.service_type_id.id,
+                        "description": self.name,
+                        "vehicle_id": self.vehicle_id.id,
+                        "amount": self.amount,
+                        "odometer": self.odometer,
+                        "vendor_id": self.vendor_id.id if self.vendor_id else False,
+                        "state": "done",
+                    }
+                )
+                .id
```
Comment (ivantodorovich) [inline fleet_vehicle_inspection/models/fleet_vehicle_inspection.py]:
```suggestion
```

💡 tip: not needed, you can assign the recordset directly

File: fleet_vehicle_inspection/tests/test_fleet_vehicle_inspection.py [hunk 3]
```diff
@@ -79,3 +80,30 @@ def test_fleet_vehicle_inspection(self):
...
             self.inspection2.button_confirm()
+
+    def test_fleet_vehicle_inspection_with_amount(self):
+        test_vendor = self.env["res.partner"].create({"name": "test vendor"})
+        self.inspection.inspection_line_ids[0].action_item_success()
+        self.inspection.inspection_line_ids[1].action_item_success()
+        self.inspection._compute_inspection_result()
+        self.inspection.amount = 100
+        self.inspection.vendor_id = test_vendor.id
+        with self.assertRaises(UserError):
```
Comment (ivantodorovich) [inline fleet_vehicle_inspection/tests/test_fleet_vehicle_inspection.py]:
```suggestion
        with self.assertRaisesRegex(UserError, "Inspection cannot be completed"):
```

💡 tip: Always use `assertRaisesRegex` over `assertRaises`, this way you are sure to be asserting the exact exception you expect. UserErrors can be arised from many different places, and so you risk to be testing something you don't want

⚠️  Also, by the look of the PR diff, it kinda looks like you were expecting a `ValidationError(msg="Must select service type")` here, not a `UserError` 🤔 is that so?

**Review by ivantodorovich — CHANGES_REQUESTED** [review]

**Review by ivantodorovich — APPROVED** [review]
/ocabot merge major

**Pull request merged**